src/algebra/group/defs.lean

```diff
@@ -446,6 +482,34 @@ class add_left_cancel_monoid (M : Type u) extends add_left_cancel_semigroup M, a
 @[protect_proj, ancestor left_cancel_semigroup monoid, to_additive add_left_cancel_monoid]
 class left_cancel_monoid (M : Type u) extends left_cancel_semigroup M, monoid M
 
+/-- A mixin for `left_cancel_monoid M`. -/
+@[protect_proj] class is_left_cancel_monoid (M : Type u) [monoid M] :=
+  (mul_left_cancel : ∀ a b c : M, a * b = a * c → b = c)
```

Comment:
I'm not sure this is necessary; you could make it a reducible def `:= is_left_cancel_semigroup M` or simply remove it and just use `is_left_cancel_semigroup`.